pkg/testrig/template/dockerfile.go

```diff
@@ -0,0 +1,34 @@
+package template
+
+func GetDockerfileStub() string {
+	return `
+FROM golang:latest as builder
+ENV http_proxy=http://docker.for.mac.host.internal:3128
+ENV https_proxy=http://docker.for.mac.host.internal:3128
+
+WORKDIR /app
+
+# allow insecure download for ANZ root CA certificates, we need them for golang
+RUN git config --global http.sslVerify false
+RUN git clone --depth 1 https://github.service.anz/ocp/ocp-cacerts.git
+RUN cp ocp-cacerts/global/*.crt /usr/local/share/ca-certificates/
+RUN update-ca-certificates
+
+# allow caching on source rebuilds
+COPY go.mod go.sum ./
+COPY vendor ./vendor
+RUN go mod download
```

Comment:
I would swap this line with the last (COPY vendor). If the go modules do not change but the vendor directory does, it will still download the modules again due to the layer caching. Or swap the COPY vendor with COPY go.mod if you think the vendor directory is less likely to change than the go.mod contents.

Reply:
`go mod download` fails if there's no vendor dir
so it always go last
but indeed, vendor dir changes less often